flake.nix

```diff
@@ -107,10 +112,17 @@
           inherit inputs self;
           pkgsFor = final;
         });
-    in {
-      # Use bleeding edge `rustfmt'.
-      rustfmt = prev.rustfmt.override {asNightly = true;};
 
+      # We depend on several nightly features of rustfmt,
+      # so pick the current nightly version.
+      # We're using `default.withComponents`
+      # which _should_ only pull the nightly rustfmt component.
+      # Alternatively, we could use nixpkgs.rustfmt,
+      # and rebuild with a (stable) fenix toolchain and `asNightly = true`,
+      # which would avoid the need to pull another channel altogether.
+      rustfmt-nightly = prev.fenix.default.withComponents ["rustfmt"];
+      rust-toolchain = final.fenix.stable;
```

Comment:
question nonblocking: how come one uses prev and one uses final?

Reply:
That was there before, technically it's to avoid infinite recursions but since if the overlay depends on something that itself added. I don't think we need this (it's been this way when I picked it up)